packages/datadog-plugin-bluebird/test/index.spec.js

```diff
@@ -3,3 +3,70 @@
 const assertPromise = require('../../dd-trace/test/plugins/promise')
 
 assertPromise('bluebird')
+
+assertPromise('bluebird', bluebird => {
+  // TODO: remove if statement when running tests only for versions ^2.11.0 and ^3.4.1
+  // https://github.com/petkaantonov/bluebird/releases/tag/v2.11.0
+  // https://github.com/petkaantonov/bluebird/releases/tag/v3.4.1
+  if (!bluebird.getNewLibraryCopy) {
+    return bluebird
+  }
+
+  return bluebird.getNewLibraryCopy()
+})
+
+describe('bluebird.getNewLibraryCopy', () => {
+  let bluebird
+  const agent = require('../../dd-trace/test/plugins/agent')
+  const plugin = require('../src')
+  const unpatch = plugin[0]['unpatch']
+  const mockIntegrations = {
+    unwrap: sinon.spy()
+  }
+
+  wrapIt()
+
+  describe('library copies to unpatch', () => {
+    withVersions(plugin, 'bluebird', version => {
+      beforeEach(() => {
+        agent.load(plugin, 'bluebird')
+      })
+
+      afterEach(() => {
+        return agent.close()
+      })
+
+      describe('without configuration', () => {
+        beforeEach(() => {
+          bluebird = require(`../../../versions/bluebird@${version}`).get()
+        })
+
+        describe('unpatching', () => {
```

Comment:
took a pass at this and had a bit of trouble doing this in a generalised way that would work for all the promise libraries, going to give this another shot with fresh eyes tomorrow.

Reply:
Ok, thanks again for the feedback @rochdev and @bengl . Regarding this last outstanding point: 

I've refactored this test to a general unpatching test for the non-native promise specs. Hit one small roadbump here which is that one of the non-native promise libraries we instrument, `promise-js` in a recent version decided to just return a native promise if it's detected.

`https://github.com/kevincennis/promise/blob/b58ef67dd4023139d0aad98ccd0cb60d8a4f9eec/src/promise.js#L7-#L13`

So it makes testing patching/unpatching a bit difficult and breaks the test for one of the versions. So i added a little one liner to skip the test in this edge case where the non-native promise library defers to native promises, sort've a punt but it started to feel out of scope for the change I was trying to test and didn't want to end up going down a rabbit hole on this.

thanks again, lmk what you think.

Reply:
Actually, if the library returns the native Promise, it should not be patched in the first place, otherwise we are double-patching it as native is already supported, so it's a bug in the `promise-js` plugin.

Can you try adding your check to `promise-js` instead? It should not affect its tests since the context is always propagated for native.

Reply:
Ok gotcha that makes sense. So what I've done is updated promise-js a bit to have slightly different plugins patching/unpatchin for versions from `0.0.3 - 0.0.6`, and versions `>=0.0.7` (when they export native promises instead if it exists per `https://github.com/kevincennis/promise/compare/0.0.6...master` ) .

 I think this is necessary for the case where the user is on the latest promises-js version but does not have native promises on their version of node, and so still would need promise-js promises instrumented.  

Additionally i've added the version filter to the promise-js specs so that we only test patching/unpatching for `0.0.3 - 0.0.6` . I suppose eventually it may make sense to have the versionFilter fine grained enough to work on a per test basis but dont think it's needed for now.

lmk what you think!